### DISCUSSION TITLE
Static site using Sveltekit adapter-static is not savable (browser save does not work)

### DISCUSSION THREAD

**Discussion by tofran** [sveltejs/kit#9502] [Q&A]
I haven't been able to make a static site using Svaltekit that is _savable_ for offline usage. _Savable_ means that the website can be saved using `ctrl + S` and be used offline. Chrome and Firefox handle this pretty well in most static apps.

## How it normally works

- A simple static website is served, with resources from other files, like `/script.js`, `/assets/image.png` and even relative paths like `./style.css`.
- The user hits `ctrl + S` and the browser prefixes all these resources with the folder name (`<PAGE_TITLE>_files`).
- The website can be opened in the browser and used offline. Usually all the form input values are cleared out, but the DOM is kept as when saved.

I have made a super simple example:

<details><summary>index.html</summary>

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Vanilla demo</title>
    <script src="/script.js"></script>
    <!-- can also be ./script.js -->
</head>
<body>
    <h1>Vanilla demo</h1>

    <label>
        Type something
        <input type="text">
    </label>
    <span>0</span> characters typed.
</body>
</html>
```

</details> 

<details><summary>script.js</summary>

```
window.addEventListener('load', () => {
    const inputElement = document.querySelector('input')
    const resultElement = document.querySelector('span')
    
    inputElement.addEventListener('keyup', () => {
        console.log(inputElement.value)
        resultElement.innerText = inputElement.value.length
    })
})
```

</details> 

When saved it creates a file `Vanilla demo.html` and a directory `Vanilla demo_files` containing `script.js`. `Vanilla demo.html` has the script source prefixed with `./Vanilla demo_files/`.

Typical working example (uses react): https://tofran.github.io/passphrase-recovery-tool/ ([source](https://github.com/tofran/passphrase-recovery-tool)).

## Sveltekit

I'm using `@sveltejs/adapter-static` as documented, and have tried to set `config.kit.paths.relative` to `true` but **not all scripts are imported with a relative prefix. Notably the ones that are dynamically added to the DOM during load.** 

![image](https://user-images.githubusercontent.com/5692603/227522314-b512848b-2be2-4ed5-abe9-b694e0395003.png)

I do not fully understand why, but this might be why both Chrome and Firefox cannot save the page correctly.

Also note that `config.kit.paths.base` cannot be `/` nor `./`, so I haven't been able to play with this setting.

I have also tried to edit the key `base` of `__sveltekit_<HASH>` in the `index.html` resulting from a build, but without success.

Live static sveltekit examples that are not savable:
https://tofran.github.io/aws-ses-smtp-credentials/ ([source](https://github.com/tofran/aws-ses-smtp-credentials)) 
https://sveltekit-static.onrender.com/ ([source](https://github.com/render-examples/sveltekit-static))

<details><summary>Vanilla example ported into Sveltekit</summary>
<p>

`npm i -D @sveltejs/adapter-static`

`echo 'export const prerender = true;' >  src/routes/+layout.js`

`src/routes/page.svelte`

```svelte
<script>
    let inputValue = ''
</script>

<label>
    <h1>Sveltekit static demo</h1>

    Type something
    <input bind:value={inputValue} type="text">
</label>
<span>{inputValue.length}</span> characters typed.
```

`svelte.config.js`

```js
import adapter from '@sveltejs/adapter-static';

const config = {
	kit: {
		adapter: adapter()
	}
};
```

</p>
</details> 

**Is this a Sveltekit limitation? If so, why? Is it documented?**

## Environment

Node: `v18.13.0`
@sveltejs/adapter-static: `2.0.1`
@sveltejs/kit:`1.13.0`
svelte: `3.57.0`
vite: `4.2.1`

**Comment by teemingc** [comment]
Those scripts have a base prefix for their URLs as part of a remote dev environment fix https://github.com/sveltejs/kit/pull/8651
Kit adds them in this manner when rendering the page.
https://github.com/sveltejs/kit/blob/5191e165b75f5616bd6b4aadf9d8a6c52e75c562/packages/kit/src/runtime/server/page/render.js#L289-L301

@benmccann is there a way to support this base path fix and the offline web archive use case that requires a relative path?
